### PR TITLE
Docs: Sidebar updates

### PR DESCRIPTION
**Pull request by joedakroub**
## Description
This PR refactors the docs sidebar to use the `SageSidebar` item levels for the heading and the nav items in order to get the active item to highlight. In addition, it adds a script to scroll the active nav item into view.

## Screenshots
|  Before  |  After  |
|--------|--------|
|Before|After|

## Testing in `sage-lib`
1. Visit http://0.0.0.0:4000/pages/component/pagination and ensure that the sidebar matches the screenshot
2. Visit other component pages and ensure that the selected item remains in the center.

## Testing in `kajabi-products`
N/A

## Related
N/A

### CODE REVIEW

File: docs/app/assets/javascripts/application.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function scrollToActiveNavItem() {
+  const activeNavItemClass = ".sage-nav__link--active";
+  const activeNavItem = document.querySelector(activeNavItemClass);
+
+  activeNavItem.scrollIntoView({ block: "center" });
```
Comment (joedakroub) [inline docs/app/assets/javascripts/application.js:5]: This scrolls the item into view and tries to ensure that it is roughly in the center of the viewport.

**Comment by joedakroub** [comment]
Hmm, I just looked at the `SageSidebar` component and noticed that these changes make the top-level nav active item have a background, which is not to spec. http://0.0.0.0:4000/pages/component/sidebar

Not sure how best to handle this. I'm open to suggestions.

File: docs/app/views/application/_sidebar.html.erb [hunk 2]
```diff
@@ -1,115 +1,118 @@
...
-        {
-          text: "Design Principles",
-          link: pages_foundation_path(:design_principles),
-          active: current_page?(pages_foundation_path(:design_principles)),
-        },
-        {
-          text: "Accessibility",
-          link: pages_foundation_path(:accessibility),
-          active: current_page?(pages_foundation_path(:accessibility)),
+        { link: "#", text: "FOUNDATIONS", items: [
```
Comment (joedakroub) [inline docs/app/views/application/_sidebar.html.erb:5]: I debated using '#' or the current page url; open to suggestions.

File: packages/sage-assets/lib/stylesheets/components/_nav.scss [hunk 3]
```diff
@@ -45,7 +45,7 @@ $-nav-icon-size: rem(20px);
   align-items: center;
   padding: rem(10px) rem(12px);
   color: $-nav-color-text;
-  border-radius: sage-border(radius);
+  border-radius: sage-border(radius-small);
```
Comment (joedakroub) [inline packages/sage-assets/lib/stylesheets/components/_nav.scss:48]: This is to match the Sage 3 spec.

File: packages/sage-assets/lib/stylesheets/components/_nav.scss [hunk 4]
```diff
@@ -56,7 +56,7 @@ $-nav-icon-size: rem(20px);
   }
 
   &:focus {
-    box-shadow: inset 0 0 0 1px $-nav-color-focus;
+    box-shadow: inset 0 0 0 2px $-nav-color-focus;
```
Comment (joedakroub) [inline packages/sage-assets/lib/stylesheets/components/_nav.scss:59]: Updated to match Sage 3 spec.

File: docs/app/views/application/_sidebar.html.erb (hunk 2 above)
Comment (monkeypox8) [reply docs/app/views/application/_sidebar.html.erb:5]: I'm not sure what the correct answer is here, but I agree that it should link somewhere. Maybe we default to the first item in the list of children?

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM, this is a great quality-of-life enhancement!

File: docs/app/assets/javascripts/application.js (hunk 1 above)
Comment (joedakroub) [reply docs/app/assets/javascripts/application.js:5]: Absolutely, I will add the `smooth` behavior in. Good call.

File: docs/app/views/application/_scripts.html.erb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 <%= javascript_include_tag "//cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js", cache: true %>
+<%= javascript_include_tag "application.js" %>
```
[the remark this answers is not in the document]
Comment (joedakroub) [reply docs/app/views/application/_scripts.html.erb:2]: Good call. I was debating on where to put this script; this seems more appropriate. 👍🏼

File: docs/app/views/application/_sidebar.html.erb (hunk 2 above)
Comment (joedakroub) [reply docs/app/views/application/_sidebar.html.erb]: I added a link to the first item in the list. 👍🏼

**Pull request merged**